src/ipc/ipc4/handler.c

```diff
@@ -940,18 +940,14 @@ ipc_cmd_hdr *ipc_compact_read_msg(void)
 
 ipc_cmd_hdr *ipc_prepare_to_send(const struct ipc_msg *msg)
 {
-	uint32_t size;
-
 	msg_data.msg_out[0] = msg->header;
-	msg_data.msg_out[1] = *(uint32_t *)msg->tx_data;
+	msg_data.msg_out[1] = msg->extension;
 
-	/* the first uint of msg data is sent by ipc data register for ipc4 */
-	size = msg->tx_size - sizeof(uint32_t);
-	if (size)
-		mailbox_dspbox_write(0, (uint32_t *)msg->tx_data + 1, size);
+	if (msg->tx_size)
+		mailbox_dspbox_write(0, (uint32_t *)msg->tx_data, msg->tx_size);
```

Comment:
I.e. this would benefit from some description in commit. Apparently the logic how to handle the special treatment of the first word, is changed in this patch. This is probably all good, but given title "header cleanup", the above code change jumps out quite a bit.